.github/workflows/pr.yml

```diff
@@ -27,37 +28,132 @@ jobs:
+    outputs:
+      run-e2e: ${{ steps.check-if-pr-has-label.outputs.run-e2e && (github.event_name != 'labeled' || (github.event_name == 'labeled' && github.event.label.name == 'ready-for-e2e')) }}
+      run-jobs: ${{ github.event_name != 'labeled' }}
+    steps:
+      - name: Get PR from branch name
+        id: check-if-pr-has-label
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const ref = '${{ github.ref }}';
+            const branch = ref.replace('refs/heads/', '');
+            console.log(ref);
+            console.log(branch);
+            const response = await github.rest.pulls.list({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open',
+              head: `${context.repo.owner}:${branch}`
+            });
+
+            if (response.data.length > 0) {
+              const pr = response.data[0];
+              const labels = pr.labels.map(label => label.name);
+              console.log('PR #', pr.number);
+              if (labels.includes('ready-for-e2e')) {
+                core.setOutput('run-e2e', true);
+                console.log('Found the label');
+              } else {
+                core.setOutput('run-e2e', false);
+                console.log('Did not find the label');
+              }
+            } else {
+              core.setOutput('run-e2e', false);
+              console.log('No PR found matching the branch');
+            }
+
   type-check:
     name: Type check
-    needs: [changes]
-    if: ${{ needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
+    needs: [changes, check-label]
+    if: ${{ needs.check-label.outputs.run-jobs == 'true' && needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
     uses: ./.github/workflows/check-types.yml
     secrets: inherit
 
   lint:
     name: Linters
-    needs: [changes]
-    if: ${{ needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
+    needs: [changes, check-label]
+    if: ${{ needs.check-label.outputs.run-jobs == 'true' && needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
     uses: ./.github/workflows/lint.yml
     secrets: inherit
 
   unit-test:
     name: Tests
-    needs: [changes]
-    if: ${{ needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
+    needs: [changes, check-label]
+    if: ${{ needs.check-label.outputs.run-jobs == 'true' && needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
     uses: ./.github/workflows/unit-tests.yml
     secrets: inherit
 
   integration-test:
     name: Tests
-    needs: [changes]
-    if: ${{ needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
+    needs: [changes, check-label]
+    if: ${{ needs.check-label.outputs.run-jobs == 'true' && needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
     uses: ./.github/workflows/integration-tests.yml
     secrets: inherit
 
+  build-api-v1:
```

Comment:
Moving all of the checks back into PR flow which enables us to disable the merge queue